tests/api/test_core.py

```diff
@@ -104,3 +105,36 @@ def test_reporting_period_to_date():
     assert start_date.minute == 0
     assert start_date.second == 0
     assert start_date.microsecond == 0
+
+
+def test_require_pagination(api_client):
+    query = """
+    query {
+        products {
+            edges {
+                node {
+                    name
+                }
+            }
+        }
+    }
+    """
+    response = api_client.post_graphql(query)
+    content = _get_graphql_content_from_response(response)
+    assert 'errors' in content
+    assert content['errors'][0]['message'] == (
+        'You must provide a `first` or `last` value to properly paginate the '
+        '`products` connection.')
+
+
+def test_total_count_query(api_client, product):
+    query = """
+    query {
+        products {
+            totalCount
+        }
+    }
+    """
+    response = api_client.post_graphql(query)
+    content = get_graphql_content(response)
+    assert content['data']['products']['totalCount'] == 1
```

Comment:
I would avoid hardcoding values.
```suggestion
    assert content['data']['products']['totalCount'] == Product.objects.count()
```